marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -469,6 +529,30 @@ handleAction settings (OpenLoginPopup intendedAction) = do
     Anonymous -> pure unit
     GithubUser -> handleAction settings intendedAction
 
+handleAction settings (ConfirmUnsavedNavigationAction intendedAction modalAction) = do
+  toConfirmUnsavedNavigation intendedAction $ ConfirmUnsavedNavigation.handleAction settings modalAction
+  handleAction settings CloseModal
+  case modalAction of
+    ConfirmUnsavedNavigation.Cancel -> pure unit
+    ConfirmUnsavedNavigation.DontSaveProject -> do
+      liftEffect $ Console.log "DontSaveProject"
+      -- FIXME: this currently is never ending loop, need to comunicate
+      -- with the preventNavigation that I want to do the action anyway
+      -- The only way I can think of at the time is to put a boolean in the
+      -- state
```

Comment:
I think we discussed that this is as good as it will get

Reply:
This was solved by making the preventNavigation a HOF